extensions/oidc/deployment/src/main/java/io/quarkus/oidc/deployment/devservices/keycloak/KeycloakDevServicesProcessor.java

```diff
@@ -220,28 +223,38 @@ public void run() {
         }
         LOG.info("Dev Services for Keycloak started.");
 
-        return prepareConfiguration(capturedDevServicesConfiguration.createRealm && startResult.createDefaultRealm,
-                startResult.realmNameToUse, devServices);
+        return new KeycloakDevServicesStartedBuildItem(
+                capturedDevServicesConfiguration.createRealm && startResult.createDefaultRealm,
+                startResult.realmNameToUse);
     }
 
-    private KeycloakDevServicesConfigBuildItem prepareConfiguration(boolean createRealm, String realmNameToUse,
-            BuildProducer<DevServicesConfigResultBuildItem> devServices) {
-        final String realmName = realmNameToUse != null ? realmNameToUse : getDefaultRealmName();
+    @BuildStep(onlyIfNot = IsNormal.class, onlyIf = { IsEnabled.class, GlobalDevServicesConfig.Enabled.class })
+    @Record(ExecutionTime.RUNTIME_INIT)
+    public KeycloakDevServicesConfigBuildItem prepareConfiguration(
+            Optional<KeycloakDevServicesStartedBuildItem> devServicesStarted,
+            BuildProducer<DevServicesConfigResultBuildItem> devServicesConfig,
+            ConfigRecorder recorder) {
```

Comment:
I don't see the recorder being used here and the non recorded code is executed at build time so I'm not sure what you did there?

Reply:
In that case the method used to record, and when it was changed someone (probably me), missed removing the recorder. I have fixed it in https://github.com/quarkusio/quarkus/pull/21325

The solution is to remove `@Record`, not to add an unused recorder.